skrf/media/circularWaveguide.py

```diff
@@ -0,0 +1,433 @@
+        '''
+        guide wavelength
+
+        the distance in which the phase of the field increases by 2 pi
+        '''
+        return 2*pi/self.beta
+
+    @property
+    def lambda_cutoff(self):
+        '''
+        cuttoff wavelength
+
+        .. math::
+            v/f
+
+         where v= 1/sqrt(ep*mu)
+        '''
+        v = 1/sqrt(self.ep*self.mu)
+        return v/self.f_cutoff
+
+    @property
+    def gamma(self):
+        '''
+        The propagation constant (aka Longitudinal wave number)
+
+        Defined as
+
+        .. math::
+
+                k_z = \\pm j \\sqrt {k_0^2 - k_c^2}
+
+        This is.
+                * IMAGINARY for propagating modes
+                * REAL  for non-propagating modes,
+
+        Returns
+        --------
+        gamma :  number
+                The propagation constant
+
+
+        '''
+        # This also holds for the circular waveguide
+        ## haringtons form
+        if False: #self.m==1 and self.n==0:
+            fs = Freespace(frequency=self.frequency, 
+                           ep_r=self.ep_r, 
+                           mu_r=self.mu_r)
+                           
+            g= where(self.f_norm>1.,
+                     sqrt(1-self.f_norm**(-2))*fs.gamma, # cutton
+                 -1j*sqrt(1-self.f_norm**(2))*fs.gamma)# cutoff
+        
+        else:
+            # TODO:  fix this for lossy ep/mu (remove abs?)
+            k0,kc = self.k0, self.kc
+            g=  1j*sqrt(abs(k0**2 - kc**2)) * (k0>kc) +\
+                    sqrt(abs(kc**2- k0**2))*(k0<kc) + \
+                    0*(kc==k0) 
+
+        g = g + self.alpha_c *(self.rho is not None)
+        
+        return g 
+        
+        
+    @property
+    def alpha_c(self):
+        '''
+        Loss due to finite conductivity and roughness of sidewalls
+
+        In units of np/m
+        See property `rho` for setting conductivity.
+
+        Effects of finite conductivity are taken from [#]_. If
+        :attr:`roughness` is not None, then its effects the conductivity
+        by
+
+
+        .. math::
+
+            \\sigma_c = \\frac{\\sigma}{k_w^2}
+
+
+        where
+
+        .. math::
+
+            k_w = 1 + e^{(-\\delta/2h)^{1.6}}
+
+            \\delta = \\mbox{skin depth}
+
+            h = \\mbox{surface roughness }
+
+
+        This is taken from Ansoft HFSS help documents.
+
+
+
+        References
+        --------------
+
+        .. [#] Chapter 9, (eq 9.8.1) of Electromagnetic Waves and Antennas by Sophocles J. Orfanidis
+        http://eceweb1.rutgers.edu/~orfanidi/ewa/
+        '''
+
+        # TODO: Implement version for circular geometry
+        raise NotImplementedError
+
+        # if self.rho is None:
+        #     return 0
+        # a,b,w,ep,rho,f_n = self.a, self.b, self.frequency.w, self.ep, \
+        #     self.rho, self.f_norm
+        # return 1./b * sqrt( (w*ep)/(2./rho) ) * (1+2.*b/a*(1/f_n)**2)/\
+        #     sqrt(1-(1/f_n)**2)
+
+    @property
+    def Z0(self):
```

Comment:
maybe it should be `z0` instead of `Z0` to be consistent with skrf conventions. tbd.